### PR TITLE
Fix article description issues

### PR DESCRIPTION
**Pull request by benhalpern**
## What type of PR is this? (check all applicable)

- [ ] Refactor
- [ ] Feature
- [x] Bug Fix
- [ ] Optimization
- [ ] Documentation Update

## Description
This PR fixes a few logical problems currently with "article description"

- Currently the attempt at a contrived description if it were blank were not resulting in anything because blank strings were being submitted and the `||` wasn't being called.
- Description wasn't present in the "internal navigation" version of the page, which _may or may not_ matter, but just in case it has any impact on SEO, I think this is the right way to go about it.
- The contrived name now includes tags at the end to help possible search engine cases.

Part of this is strictly for hypothetical SEO issues, part is slightly more for user-reading purposes where we use the description in different places where now it's erroneously not present.

### CODE REVIEW

File: app/models/article.rb [hunk 1]
```diff
@@ -273,6 +273,14 @@ def self.trigger_index(record, remove)
     end
   end
 
+  def contrived_description
```
Comment (rhymes) [inline app/models/article.rb]: can we call this `artificial_description` ? I had to google what `contrived` means 😅

#nonnativespeakersproblems

File: app/decorators/article_decorator.rb [hunk 2]
```diff
@@ -51,4 +51,12 @@ def internal_utm_params(place = "additional_box")
   def published_at_int
     published_at.to_i
   end
+
+  def description_and_tags
+    modified_description = description.strip
+    modified_description += "." unless description.end_with?(".")
+    return modified_description unless  cached_tag_list.present?
```
Comment (rhymes) [inline app/decorators/article_decorator.rb]: extra space after `unless`

File: app/models/article.rb [hunk 3]
```diff
@@ -493,7 +502,7 @@ def evaluate_front_matter(front_matter)
     self.published_at = parse_date(front_matter["date"]) if published
     self.main_image = front_matter["cover_image"] if front_matter["cover_image"].present?
     self.canonical_url = front_matter["canonical_url"] if front_matter["canonical_url"].present?
-    self.description = front_matter["description"] || description || "#{body_text[0..80]}..."
+    self.description = front_matter["description"] if front_matter["description"].present? || front_matter["title"].present? # Do this if frontmatte exists at all
```
Comment (rhymes) [inline app/models/article.rb]: small typo, `frontmatter`

File: app/models/article.rb (hunk 1 above)
Comment (benhalpern) [reply app/models/article.rb]: Changed to `processed_description`

Comment (rhymes) [reply app/models/article.rb]: thanks!

**Pull request merged**